airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/cursor/default_concurrent_file_based_cursor.py

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from airbyte_cdk.models import AirbyteMessage, AirbyteStateMessage, AirbyteStateType, AirbyteStream, AirbyteStreamState, StreamDescriptor
+from airbyte_cdk.models import Type as MessageType
+from airbyte_cdk.sources.file_based.config.file_based_stream_config import FileBasedStreamConfig
+from airbyte_cdk.sources.file_based.stream.cursor import DefaultFileBasedCursor
+from airbyte_cdk.sources.message import MessageRepository
+from airbyte_cdk.sources.streams.concurrent.cursor import Cursor
+from airbyte_cdk.sources.streams.concurrent.partitions.partition import Partition
+from airbyte_cdk.sources.streams.concurrent.partitions.record import Record
+
+
+class DefaultConcurrentFileBasedCursor(DefaultFileBasedCursor, Cursor):
+    def __init__(
+        self,
+        stream: AirbyteStream,
+        stream_config: FileBasedStreamConfig,
+        message_repository: MessageRepository,
+    ) -> None:
+        self.stream = stream
+        self._most_recent_record = None
+        self._message_repository = message_repository
+        DefaultFileBasedCursor.__init__(self, stream_config)
+
+    def observe(self, record: Record) -> None:
+        pass  # TODO: do we need this to do something?
```

Comment:
This method is mostly for streams that do not slice the records using datetime. I think this will not be used for S3 and we can rely entirely on `close_partition`